app/utils/parse-form-data.server.ts

```diff
@@ -97,3 +97,21 @@ export const withValidatedFormData =
     ...rest,
     data: await parseFormData(schema, request),
   });
+
+/**
+ * A function that validates the form data from a request against a provided Zod Schema
+ *
+ *
+ * @param schema - The Zod schema to validate the form data against.
+ * @param request - The request with form data to be parsed.
+ * @returns The middleware object with the parsed form data.
+ * @throws Will throw a 400 Bad Request error if the form data does not conform
+ * to the schema with an  object containing the formatted error details.
+ */
+export const validateFormDataFromRequest = async <Schema extends ZodType>({
+  request,
+  schema,
+}: {
+  request: Request;
+  schema: Schema;
+}) => await parseFormData(schema, request);
```

Comment:
This just seems like an object wrapper around `parseFormData`. Should we refactor `parseFormData` to be this object wrapper instead?

Reply:
This comment seems unaddressed 🤔

Reply:
I missed this comment, I would redact this feature